src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -189,12 +189,8 @@ class Peer(
             }
     }
 
-    val finalWallet = ElectrumMiniWallet(nodeParams.chainHash, watcher.client, scope, nodeParams.loggerFactory, name = "final")
-    val finalAddress: String = nodeParams.keyManager.finalOnChainWallet.address(addressIndex = 0L).also { finalWallet.addAddress(it) }
-
-    val swapInWallet = ElectrumMiniWallet(nodeParams.chainHash, watcher.client, scope, nodeParams.loggerFactory, name = "swap-in")
-    val legacySwapInAddress: String = nodeParams.keyManager.swapInOnChainWallet.legacySwapInProtocol.address(nodeParams.chain).also { swapInWallet.addAddress(it) }
-    val swapInAddress: String = nodeParams.keyManager.swapInOnChainWallet.swapInProtocol.address(nodeParams.chain).also { swapInWallet.addAddress(it) }
+    val finalWallet = FinalWallet(nodeParams.chain, nodeParams.keyManager.finalOnChainWallet, watcher.client, scope, nodeParams.loggerFactory)
+    val swapInWallet = SwapInWallet(nodeParams.chain, nodeParams.keyManager.swapInOnChainWallet, watcher.client, addressGenerationWindow = 20, scope, nodeParams.loggerFactory)
```

Comment:
Notice the default address generation window.

Reply:
IIUC it works a bit differently from other wallets: we will first scan `addressGenerationWindow` addresses but then unused addresses will be scan one by one (and not in blocks of `addressGenerationWindow` addresses)

Reply:
The last version ([5c4cbcb](https://github.com/ACINQ/lightning-kmp/pull/584/commits/5c4cbcb5973f65488a5f74854f493c32067bed3f)) removes this window altogether since we are simply iterating "synchronously" over addresses until we find the first unused one.